logtail/logtail.go

```diff
@@ -757,6 +767,33 @@ func (l *Logger) Write(buf []byte) (int, error) {
 	return inLen, err
 }
 
+func redactIPs(buf []byte) []byte {
+	regexMatchesIPv6 := regexp.MustCompile(`([0-9a-fA-F]{1,4}):([0-9a-fA-F]{1,4}):([0-9a-fA-F:]{1,4})*`)
+	regexMatchesIPv4 := regexp.MustCompile(`(\d{1,3})\.(\d{1,3})\.\d{1,3}\.\d{1,3}`)
```

Comment:
We should bump these variables up to the package level so we don't re-compile the patterns each time it's called.
```suggestion
var (
	regexMatchesIPv6 = regexp.MustCompile(`([0-9a-fA-F]{1,4}):([0-9a-fA-F]{1,4}):([0-9a-fA-F:]{1,4})*`)
	regexMatchesIPv4 = regexp.MustCompile(`(\d{1,3})\.(\d{1,3})\.\d{1,3}\.\d{1,3}`)
)

func redactIPs(buf []byte) []byte {
```